rollout/sync.go

```diff
@@ -378,6 +378,12 @@ func (c *rolloutContext) scaleReplicaSet(rs *appsv1.ReplicaSet, newScale int32,
 			revision, _ := replicasetutil.Revision(rs)
 			c.recorder.Eventf(rollout, record.EventOptions{EventReason: conditions.ScalingReplicaSetReason}, conditions.ScalingReplicaSetMessage, scalingOperation, rs.Name, revision, oldScale, newScale)
 			c.replicaSetInformer.GetIndexer().Update(rs)
+			if rollout.Spec.WorkloadRef != nil && rollout.Spec.WorkloadRef.ScaleDown == v1alpha1.ScaleDownProgressively {
```

Comment:
I realized that for the `ScaleDownProgressively` option the deployment will be scaled down without waiting  for the rollout replicas to become available.
Do you have any suggestions where we can test that? We should scale down the deployment only after the replicas become available.
cc: @carlossg

Reply:
I changed the logic for the `progressively` scaleDown option, now the deployment is scaled down as the rollout is scaled up and its replicas become ready.